pkg/storkctl/schedulepolicy.go

```diff
@@ -60,6 +67,214 @@ func newGetSchedulePolicyCommand(cmdFactory Factory, ioStreams genericclioptions
 	return getSchedulePolicyCommand
 }
 
+func newCreateSchedulePolicyCommand(cmdFactory Factory, ioStreams genericclioptions.IOStreams) *cobra.Command {
+	var schedulePolicyType string
+	var intervalMinutes int
+	var retain int
+	var time string
+	var dailyForceFullSnapshotDay string
+	var dayOfWeek string
+	var dateOfMonth int
+
+	createSchedulePolicyCommand := &cobra.Command{
+		Use:     schedulePolicySubcommand,
+		Aliases: schedulePolicyAliases,
+		Short:   "Create schedule policy",
+		Run: func(c *cobra.Command, args []string) {
+			if len(args) != 1 {
+				util.CheckErr(fmt.Errorf("exactly one name needs to be provided for schedule policy name"))
+				return
+			}
+			schedulePolicyName := args[0]
+			var policyItem storkv1.SchedulePolicyItem
+			switch schedulePolicyType {
+			case "Interval":
+				var intervalPolicy storkv1.IntervalPolicy
+				intervalPolicy.IntervalMinutes = intervalMinutes
+				if retain == 0 {
+					intervalPolicy.Retain = storkv1.DefaultIntervalPolicyRetain
+				} else {
+					intervalPolicy.Retain = storkv1.Retain(retain)
+				}
+				// Validate the user input
+				err := intervalPolicy.Validate()
+				if err != nil {
+					util.CheckErr(err)
+					return
+				}
+				policyItem.Interval = &intervalPolicy
+			case "Daily":
```

Comment:
Remind me why are we making thie Daily and not "daily". 
Similarly we should have weekly, interval and monthly.
Output Yaml anyway gets created in lowercase. We should match the create command.
 Better even would be that we convert user input to lowercase so we can become case insensitive wrt the user input.
```type SchedulePolicyItem struct {
	// Interval policy that will be triggered at the specified interval
	Interval *IntervalPolicy `json:"interval"`
	// Daily policy that will be triggered daily at a specified time
	Daily *DailyPolicy `json:"daily"`
	// Weekly policy that will be triggered on the specified day of the week at
	// the specified time
	Weekly *WeeklyPolicy `json:"weekly"`
	// Monthly policy that will be triggered on the specified date of the month
	// at the specified time
	Monthly *MonthlyPolicy `json:"monthly"`
}

Reply:
I had just followed the convention from existing code
```
const (
	// SchedulePolicyTypeInvalid is an invalid schedule policy
	SchedulePolicyTypeInvalid SchedulePolicyType = "Invalid"
	// SchedulePolicyTypeInterval is the type for an interval schedule policy
	SchedulePolicyTypeInterval SchedulePolicyType = "Interval"
	// SchedulePolicyTypeDaily is the type for a daily schedule policy
	SchedulePolicyTypeDaily SchedulePolicyType = "Daily"
	// SchedulePolicyTypeWeekly is the type for a weekly schedule policy
	SchedulePolicyTypeWeekly SchedulePolicyType = "Weekly"
	// SchedulePolicyTypeMonthly is the type for a monthly schedule policy
	SchedulePolicyTypeMonthly SchedulePolicyType = "Monthly"
)

// GetValidSchedulePolicyTypes returns the valid types of schedule policies that
// can be configured
func GetValidSchedulePolicyTypes() []SchedulePolicyType {
	return []SchedulePolicyType{SchedulePolicyTypeInterval, SchedulePolicyTypeDaily, SchedulePolicyTypeWeekly, SchedulePolicyTypeMonthly}
}
```
However we can make the change to convert it to lowercase as I am no longer using this GetValidSchedulePolicyTypes method in my code.

Reply:
Addressed in the revision